### DISCUSSION TITLE
Undistort Images

### DISCUSSION THREAD

**Discussion by aaronlsmiles** [decadenza/SimpleStereo#15] [Q&A] [answered]
In the code of the examples _004a UndistortImages.py_ file, the `imgPath` points to a couple of calibration images.

In practice we would not do this though, correct? In practice we would use this code to undistort images of scenes, not the calibration images, or am I misunderstanding the pipeline?

**Comment by decadenza** [comment, marked as answer]
Nope, the `imgPath` defined [here](https://github.com/decadenza/SimpleStereo/blob/59dc846b36f9aadfd29eb290e48be048fdb04d50/examples/004a%20UndistortImages.py#L13) is just the directory.

Yes, you can undistort any image taken from the camera _after_ calibration. Then you can undistort any scenes, including the calibration ones!

Normal pipeline is:

1. calibrate
2. rectify
3. take left and right scene pictures
4. stereo matching
5.  triangulation and 3D reconstruction

**Reply by aaronlsmiles** [reply]
Where does Undistort fit in that pipeline though? I'm assuming it must be done before step 4 (stereo matching)?

**Reply by decadenza** [reply]
Is optional after 3... You may need undistorted images.
Otherwise distortion is automatically corrected during triangulation.
